### PR TITLE
Refix UI hit chance

### LINKED ISSUE #53124: Setting a default firing mode breaks "moves to fire" calculation
### Describe the bug

If you set a default firing mode through f -> F, the "moves to fire" listed in the info box become too low. The mode you select (and all slower ones) is set to the same as "immediate", and the others seem to then be calculated based on that incorrect values. This seems to be a visual bug that does not affect the actual speed of firing.

### Steps to reproduce

1. Equip a gun
2. Press f->F and select a default firing mode

### Expected behavior

The "moves to fire" listing should still be accurate when you have a default firing mode.

### Screenshots

image
Precise as default

image
Regular as default

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: d8a51aa [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    Bionic Slots [cbm_slots],
    Extra Mutated Scenarios [extra_mut_scens]
]

### Additional context

Setting a default firing mode also disables the "immediate" mode that's normally the default, which is annoying.

### LINKED ISSUE #59492: Aiming is broken
### Describe the bug

Before [Cataclysm-DDA experimental build 2022-07-22-0211](https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/cdda-experimental-2022-07-22-0211) I can see my current hit chance, which are increasing when I am pressing "."
After this update I cannot. And it VERY uncomfortable, becouse I dont know, can I shoot enemy or not

I have also two videos, but I dont know, how to upload it becouse of 10 mb of file limit, if you show me, how to do it - I will do it

### Steps to reproduce

1. Load [Cataclysm-DDA experimental build 2022-07-22-0050](https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/cdda-experimental-2022-07-22-0050)
2. Make with debud meny any gun, find enemy, press f (fire wielded weapon), than press "." ("to steady tour aim")
3. See your current hit chance
4. Load [Cataclysm-DDA experimental build 2022-07-22-0211](https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/cdda-experimental-2022-07-22-0211)
5. Make the same steps
6. See only default hit chances, not current
7. Pray to the god of aim and cry

### Expected behavior

I want to know, can I shoot this dead punks or not

### Screenshots

On the screenshots below I pressed steady aim button "." about ~ 16 time on bouth screens, but the difference in hit chance is big
And you can see on first screen "curent aim", which are VERY usefull in game
Before [Cataclysm-DDA experimental build 2022-07-22-0211](https://github.com/CleverRaven/Cataclysm-DDA/releases/tag/cdda-experimental-2022-07-22-0211)
image
After
image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19044.1826 (21H2)
- Game Version: 7fe8d5b [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

_No response_

### LINKED ISSUE #59564: Immediate aiming mode displays incorrect values
### Describe the bug

Increasing steadiness doesn't affect the percentage display of immediate aim. Only a visual issue, the action will complete with the expected accuracy.

### Steps to reproduce

1. Target an enemy
2. Increase steadiness
3. Displayed values are not impacted by steadiness.

### Expected behavior

Percentage values should adjust based upon steadiness.

### Screenshots

_No response_

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19042.1766 (20H2)
- Game Version: c2b61d6 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Blaze Industries [blazeindustries],
    Aftershock [aftershock],
    Xedra Evolved [xedra_evolved],
    No Rail Stations [No_Rail_Stations],
    Alternative Map Key [alt_map_key]
]

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by irwiss**
#### Summary
None

#### Purpose of change

Fixes #59492
Fixes #59564
Re-fix #53124

#### Describe the solution

TLDR; Stripped to just one code branch, isolated hit chance calculations into their own methods and printing code to their own methods, then tried my best restoring the stripped code... Which turned out nobody is using for a month anyway :(

Tried const-ing and converting to references everywhere and tidying up the code into isolated chunks, simulation should no longer do weird tricks modifying Character to run the cycle. Unfortunately first commit turned out to be super hairy as I had changes all over the place to untangle and took a while before I had a "complete" commit.

May be these changes will help writing a more decent widget for aiming, the visuals differ slightly compared to before the commits but changes should now be trivial - everything needed to draw any UI is calculated into a single struct and printing from there should be much easier.

#### Describe alternatives you've considered

PR a revert on my commit and vow to never touch panel code ever again

#### Testing

Spawn self + gun, spawn humvee tow nearby, spawn a few zeds
`fF` and select "immediate" for default aiming mode - that should show you 'current aim' as first item, with a tiny amount of moves and superlow steadiness/hit chances.
If you spam/hold `.` it should spend moves on aiming and eventually surpass the next aiming mode, and pressing f will fire at that aiming level.
Alternatively `fF` and select for example "careful" mode - you'll see "current" aim jumps over it and shows the moves your character will spend to fire if you press "f", you may still fine tune aiming by spam/holding `.` until it passes "precise" and so on.
Aiming with `.` works until you cap out at 100% steadiness, and there should no longer be improvement in steadiness (but chances may rise/fall according to zombies getting closer/away from the character)

Tested shooting at zombies - for example shoot a shot that costs 300 moves - see that roughly 3 turns have passed
Tested throwing stuff - as that has some code that got touched during the refactor
Tested shooting from the humvee tow controls to check vehicle aiming still works
Tested shooting arrows in similar way

#### Additional context

https://user-images.githubusercontent.com/6560075/180575582-189dcadb-fd9a-4474-9591-600d4450a783.mp4

### CODE REVIEW

**Comment by RelMayers** [comment]
Have you checked how graphic presentation of accuracy looks like?

**Comment by irwiss** [comment]
> Have you checked how graphic presentation of accuracy looks like?

@RelMayers If you mean the numbers / "bars" styling I adjusted it now, if that's not what you mean then please explain

image

Also fixed the

**Comment by KHeket** [comment]
I dont know how github programming magick works, but can you boost this pr and make it quicker, please. Becouse every time I need to shoot big zombie groups - its make me feel very big pain. Or may be I can somehow help with my own? But I dont have programming magick skill yet

**Comment by pjf** [comment]
I've got this applied on my local branch and it's working great. However I'm used to the old UI for showing great/normal/graze odds, where they're all symbols in a bar.

Does this PR change that, or did that change elsewhere and I've only recently noticed?

Many thanks!

**Comment by irwiss** [comment]
@pjf No that's not intended, the last commit should make it behave fairly close to what it behaved before with "symbols", if you have the time to playtest

**Comment by Night-Pryanik** [comment]
> Does this PR change that, or did that change elsewhere and I've only recently noticed?

This is changed in game options, in Interface tab. It's called `Aim window display style`.

**Comment by pjf** [comment]
> This is changed in game options, in Interface tab. It's called Aim window display style.

Today I learned this is a thing!

However...

This is how the aim window used to look with bars (note with `*`, `|` and `+`):

aim-before

And this is how it looks now (note the lack of symbols, and numbers being used to communicate crit/hit/graze):

aim

**Comment by irwiss** [comment]
@pjf https://github.com/CleverRaven/Cataclysm-DDA/pull/59512/commits/05deeeefead22eaf493fedd59595689cc7ef90ec This commit should make it similar to what you have in screenshot

image

**Comment by pjf** [comment]
Oops, sorry! Didn't have the most recent commits on this branch. Recompiling now.

**Comment by pjf** [comment]
Just reporting back in to say that with the latest changes this works great! Thank you *so much*!
